File: ewbapp/Models/Enums/LantanaVariant.swift
import SwiftUI

enum LantanaVariant: String, CaseIterable, Codable {
    case pink = "pink"
    case red = "red"
    case pinkEdgedRed = "pinkEdgedRed"
    case orange = "orange"
    case white = "white"
    case unknown = "unknown"

    var displayName: String {
        switch self {
        case .pink: return "Pink"
        case .red: return "Red"
        case .pinkEdgedRed: return "Pink-Edged Red"
        case .orange: return "Orange"
        case .white: return "White"
        case .unknown: return "Unknown"
        }
    }

    var color: Color {
        switch self {
        case .pink: return Color(red: 1.0, green: 0.41, blue: 0.71)
        case .red: return Color(red: 0.86, green: 0.08, blue: 0.24)
        case .pinkEdgedRed: return Color(red: 0.9, green: 0.25, blue: 0.45)
        case .orange: return Color(red: 1.0, green: 0.55, blue: 0.0)
        case .white: return Color(red: 0.95, green: 0.95, blue: 0.95)
        case .unknown: return Color.gray
        }
    }

    var controlMethods: [TreatmentMethod] {
        switch self {
        case .pink: return [.foliarSpray, .splatGun]
        case .red: return [.cutStump, .basalBark]
        case .pinkEdgedRed: return [.cutStump, .foliarSpray]
        case .orange: return [.foliarSpray, .basalBark]
        case .white: return [.foliarSpray, .splatGun]
        case .unknown: return [.foliarSpray]
        }
    }

    var distinguishingFeatures: String {
        switch self {
        case .pink: return "Soft pink flowers, often fading to yellow centres. Common in disturbed areas."
        case .red: return "Deep red-orange flowers. Most aggressive spreader. Toxic berries."
        case .pinkEdgedRed: return "Pink outer petals with red centre. Hybrid characteristics."
        case .orange: return "Bright orange-yellow flowers. Common along watercourses."
        case .white: return "White to cream flowers. Less vigorous but still invasive."
        case .unknown: return "Variant not clearly identifiable. Treat as per nearest match."
        }
    }

    var hasBiocontrolConcern: Bool { self == .pink }
}

File: app/src/main/java/org/yac/llamarangers/domain/enums/LantanaVariant.kt
package org.yac.llamarangers.domain.enums

import androidx.compose.ui.graphics.Color

enum class LantanaVariant(
    val displayName: String,
    val color: Color,
    val distinguishingFeatures: String,
    val controlMethods: List<TreatmentMethod>
) {
    PINK(
        displayName = "Pink",
        color = Color(0xFFFF69B5),
        distinguishingFeatures = "Pale pink flowers with yellow centre. Soft stem hairs. Wet season: watch for lantana bug damage.",
        controlMethods = listOf(TreatmentMethod.FOLIAR_SPRAY, TreatmentMethod.BASAL_BARK)
    ),
    RED(
        displayName = "Red",
        color = Color(0xFFDC143C),
        distinguishingFeatures = "Bright red/orange flowers. Woody stems. Most common variant in Port Stewart region.",
        controlMethods = listOf(TreatmentMethod.CUT_STUMP, TreatmentMethod.SPLAT_GUN)
    ),
    PINK_EDGED_RED(
        displayName = "Pink-Edged Red",
        color = Color(0xFFE64073),
        distinguishingFeatures = "Red flowers with distinct pink outer petal edges. Intermediate growth habit.",
        controlMethods = listOf(TreatmentMethod.FOLIAR_SPRAY, TreatmentMethod.CUT_STUMP)
    ),
    ORANGE(
        displayName = "Orange",
        color = Color(0xFFFF8C00),
        distinguishingFeatures = "Orange-yellow flowers. Sprawling growth. Often found along creek lines.",
        controlMethods = listOf(TreatmentMethod.BASAL_BARK, TreatmentMethod.FOLIAR_SPRAY)
    ),
    WHITE(
        displayName = "White",
        color = Color(0xFFF0F0F0),
        distinguishingFeatures = "White flowers, sometimes with pale yellow. Less aggressive than red/pink variants.",
        controlMethods = listOf(TreatmentMethod.FOLIAR_SPRAY)
    ),
    UNKNOWN(
        displayName = "Unknown",
        color = Color(0xFF888888),
        distinguishingFeatures = "Variant not identified. Record photo for later classification.",
        controlMethods = listOf(TreatmentMethod.FOLIAR_SPRAY)
    );

    companion object {
        fun fromRaw(raw: String): LantanaVariant =
            entries.firstOrNull { it.name == raw } ?: UNKNOWN
    }
}
